src/SharedWeb/Utilities/ServiceCollectionExtensions.cs

```diff
@@ -177,6 +181,22 @@ public static void AddDefaultServices(this IServiceCollection services, GlobalSe
         // Required for HTTP calls
         services.AddHttpClient();
 
+        // Add open telemetry
+        var diagnosticsConfig = DiagnosticsConfig.For(globalSettings);
+        services.AddSingleton(diagnosticsConfig.ActivitySource);
+        services.AddMiddlewareAnalysis();
+        services.Insert(0, ServiceDescriptor.Transient<IStartupFilter, AnalysisStartupFilter>()); // Insert at beginning to catch all middleware
```

Comment:
⚠️ Your notes indicate the middleware analysis dependency is a little flaky -- what are you interested in gathering here? I wasn't personally aware of this library.

Reply:
I was concerned at the time particularly with our own current context middleware. In its pure state it would use JWTs exclusively to build context for the api call, but it's been expanded quite a lot and is making db calls.

I didn't see anything concerning, but it's definitely possible given more complex users than in my dev systems.

from what I saw it's flaky in that it doesn't always execute the same middleware in similar requests -- it's possible there's some asp.net magic I don't understand that's optimizing middleware executions out.